jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppContext.java

```diff
@@ -1228,11 +1240,11 @@ protected void stopContext() throws Exception
         }
         finally
         {
-            if (!(_initialClassLoader instanceof WebAppClassLoader))
+            ClassLoader loader = getClassLoader();
+            if (loader != _initialClassLoader)
             {
-                ClassLoader loader = getClassLoader();
-                if (loader instanceof URLClassLoader)
-                    ((URLClassLoader)loader).close();
+                if (loader instanceof URLClassLoader urlClassLoader)
+                    urlClassLoader.close();
```

Comment:
Wouldn't this break a restart of the context?

Reply:
That's a good question... But existing behavior

Reply:
@lachlan-roberts I cleaned this up.  Can you re-review?

Reply:
Not sure I understand the purpose of this... Can you explain the logic behind this change?

`loader` will always be a `WebAppClassLoader` so this will never be true, unless there is an overridden `configureClassLoader` configured to return a `URLClassLoader`. If this is the case can't they also override `stopContext` and close it themselves?

Reply:
@lachlan-roberts `WebAppClassLoader` is-a `URLClassLoader`, so this will always be true when the current loader is not the initial loader (i.e. we created the loader during doStart).   So probably could assume that, but instanceof check is nice way to do the caste without a warning.